### PR TITLE
Try using helix sdk support directly again

### LINKED ISSUE #23575: [Helix] Need to stop permanently modifying path on windows
Related to https://github.com/dotnet/core-eng/issues/10185

We made the paths get too long so dotnet-install stopped working on machines that don't get reimaged frequently, need to fixup runtests.cmd to stop permanently modifying the path

@pranavkm @MattGal

### PR DESCRIPTION
**Pull request by HaoK**
@MattGal @pranavkm

### CODE REVIEW

File: eng/helix/helix.proj
Comment (HaoK) [inline eng/helix/helix.proj:27]: Okay looks like hardcoding gets us almost there, @dougbu do you know why $(NETCoreSdkVersion) isn't defined in helix.proj, or I guess more importantly, what's the best way to flow that in, do we have access to that variable or the sdk version in yaml so I guess we can just explicitly pass it into the helix jobs if so?

Comment (MattGal) [reply eng/helix/helix.proj:27]: Looking at your Yaml, if you trust that $(NETCoreSdkVersion) you were talking about yesterday is set when that build is invoked (it was hard to follow its flow) you can really really force the issue on the command line that builds your helix .proj with a simple /p:DotNetCliVersion=$(NETCoreSdkVersion)

Comment (dougbu) [reply eng/helix/helix.proj:27]: I don't know where `$(NETCoreSdkVersion)` comes from though I suspect it's in the Helix SDK somewhere. Would check there.

Comment (MattGal) [reply eng/helix/helix.proj:27]: I believe it actually comes from the .NET Core SDK being used, not Arcade, via Microsoft.NETCoreSdk.BundledVersions.props.

Comment (HaoK) [reply eng/helix/helix.proj]: Okay so I guess that's the issue, since helix.proj is using the Helix.SDK project, NETCoreSdkVersion isn't defined since its coming from `Sdk="Microsoft.Net.sdk"` Its also not in the yaml, so feeding it in explicitly doesn't work.  Is there an another easy way for me to import some target from the sdk to get the SDK version defined in our helix.proj?

Comment (pranavkm) [reply eng/helix/helix.proj]: You could consider only importing the Microsoft.NET.Sdk's props file

```xml
<Import project="Sdk.props" Sdk="Microsoft.NET.Sdk" />
```

Comment (MattGal) [reply eng/helix/helix.proj]: There's a few ways to do this, but I think it's mostly done through global.json like here: https://github.com/dotnet/runtime/blob/master/global.json#L7-L9 which brings along and uses that SDK version, resulting in that property being set to that value, by that dotnet building thing (but for runtime they have special stuff to use the runtime they've built, already)

Comment (HaoK) [reply eng/helix/helix.proj]: Cool I actually just tried pushing that actually, including the sdk.props

**Comment by HaoK** [comment]
@Tratcher @jkotalik looks like all of the IIS functional tests are failing on this PR, but everything else is working, can you guys take a look and see if its obvious what's going on? The main difference in this PR is we switched to using the build in helix sdk support, so its possible some of the paths and side effects are slightly different, not sure if IIS functionals are relying on anything there...

**Comment by HaoK** [comment]
Direct link to a build with the failing tests, since I'm going to fix a build warning in the PR:

https://dev.azure.com/dnceng/public/_build/results?buildId=715506&view=ms.vss-test-web.build-test-results-tab&runId=22171954&paneView=debug

File: eng/helix/helix.proj
Comment (HaoK) [reply eng/helix/helix.proj]: Looks like that works, but it does introduce a build warning (which is treated as an error), any ideas how to stop the double import?

```
 [error].dotnet\sdk\5.0.100-preview.6.20310.4\Sdks\Microsoft.NET.Sdk\Sdk\Sdk.props(40,3): error MSB4011: "F:\workspace\_work\1\s\.dotnet\sdk\5.0.100-preview.6.20310.4\Current\Microsoft.Common.props" cannot be imported again. It was already imported at "F:\workspace\_work\1\s\.packages\microsoft.dotnet.helix.sdk\5.0.0-beta.20316.1\tools\Microsoft.DotNet.Helix.Sdk.props (3,3)". This is most likely a build authoring error. This subsequent import will be ignored. [F:\workspace\_work\1\s\eng\helix\helix.proj]
```

Comment (dougbu) [reply eng/helix/helix.proj]: The Helix SDK (tools\Microsoft.DotNet.Helix.Sdk.props) imports Microsoft.Common.props and that's unconditional. Unfortunately, you'll need to suppress the warning using `<NoWarn>` or `<WarningsNotAsErrors>`.

Longer term, it might make sense to add a property in the Helix SDK to control the relevant `<Import>`.

Comment (MattGal) [reply eng/helix/helix.proj]: https://github.com/dotnet/arcade/pull/5747/files

Comment (dougbu) [reply eng/helix/helix.proj]: @HaoK please file an issue to remove whichever suppression you choose once @MattGal's longer-term fix is in and we get a new Helix SDK.

File: eng/helix/helix.proj
Comment (dougbu) [inline eng/helix/helix.proj:28]: @alexperovich is looking for ways to get this working without importing the .NET SDK's Sdk.props file. What we're doing here is apparently unsupported and suppressing the warning will likely lead to other issues.

Comment (dougbu) [reply eng/helix/helix.proj:28]: Suggest waiting for @alexperovich to get dotnet/arcade#5748 in. At that point, you can remove this line and either wait for the next Arcade promotion to `.NET Eng - Latest` or update the Arcade dependencies from `.NET Eng - Validation` (short-cutting a validation ring). I slightly prefer the latter because Arcade validation of our repo doesn't run tests. But, I'm not sure we should merge this PR without also doing Arcade validation.

Comment (dougbu) [reply eng/helix/helix.proj:28]: BTW with @alexperovich's change, you also won't need the added `<Import>` [above](https://github.com/dotnet/aspnetcore/pull/23585/files#diff-500537596960313988af742d444b490dR2) nor the `$(DotNetCliPackageType)` setting.

Comment (HaoK) [reply eng/helix/helix.proj:28]: Cool that's fine, we need to still track down why IIS functionals are no longer working with this change, so sounds like this will mostly just work once that PR is in

Comment (dougbu) [reply eng/helix/helix.proj:28]: I recommend removing these two lines as well as the `<Import>`. The Helix SDK will default to using the SDK from the build when it submits the job, no reason to duplicate what it's doing.

Comment (HaoK) [reply eng/helix/helix.proj:28]: We won't see those changes immediately though right?  We need to get our side working irrespective of the helix sdk improvements.

File: eng/helix/content/runtests.cmd
Comment (jkotalik) [inline eng/helix/content/runtests.cmd:20]: This is probably breaking IIS tests. 

The error coming from IIS is:
```
[301.828s] [Microsoft.AspNetCore.Server.IntegrationTesting.IIS.IISDeployer] [Information] Could not find dotnet.exe at 'C:\h\w\AED609D4\p\sdk\dotnet.exe'' 
[301.828s] [Microsoft.AspNetCore.Server.IntegrationTesting.IIS.IISDeployer] [Information] End Event Log Message.
[301.828s] [Microsoft.AspNetCore.Server.IntegrationTesting.IIS.IISDeployer] [Information] [aspnetcorev2.dll] InvalidOperationException 'Could not find dotnet.exe at 'C:\h\w\AED609D4\p\sdk\dotnet.exe'' caught at F:\workspace\_work\1\s\src\Servers\IIS\AspNetCoreModuleV2\CommonLib\HostFxrResolutionResult.cpp:64 
[301.828s] [Microsoft.AspNetCore.Server.IntegrationTesting.IIS.IISDeployer] [Information] [aspnetcorev2.dll] Failed HRESULT returned: 0x8007023e at F:\workspace\_work\1\s\src\Servers\IIS\AspNetCoreModuleV2\AspNetCore\HandlerResolver.cpp:68 
[301.828s] [Microsoft.AspNetCore.Server.IntegrationTesting.IIS.IISDeployer] [Information] [aspnetcorev2.dll] Failed HRESULT returned: 0x8007023e at F:\workspace\_work\1\s\src\Servers\IIS\AspNetCoreModuleV2\AspNetCore\HandlerResolver.cpp:171 
[301.828s] [Microsoft.AspNetCore.Server.IntegrationTesting.IIS.IISDeployer] [Information] [aspnetcorev2.dll] Failed HRESULT returned: 0x8007023e at F:\workspace\_work\1\s\src\Servers\IIS\AspNetCoreModuleV2\AspNetCore\applicationinfo.cpp:181 
[301.829s] [Microsoft.AspNetCore.Server.IntegrationTesting.IIS.IISDeployer] [Information] [aspnetcorev2.dll] Failed HRESULT returned: 0x8007023e at F:\workspace\_work\1\s\src\Servers\IIS\AspNetCoreModuleV2\AspNetCore\applicationinfo.cpp:96 
[301.829s] [Microsoft.AspNetCore.Server.IntegrationTesting.IIS.IISDeployer] [Information] [aspnetcorev2.dll] Event Log: 'Failed to start application '/LM/W3SVC/1/ROOT', ErrorCode '0x8007023e'.' 
```

I think it may be using DOTNET_HOME to find dotnet, I think you'll need to update https://github.com/dotnet/aspnetcore/blob/9a1810c1dbe432fc7bc7e8bc68fa22ab787c0452/src/Hosting/Server.IntegrationTesting/src/Common/DotNetCommands.cs.

Comment (HaoK) [reply eng/helix/content/runtests.cmd:20]: Ah cool that's why I didn't find it, i was searching under servers not hosting... thanks for pointing this out

Comment (HaoK) [reply eng/helix/content/runtests.cmd:20]: I'm going to try removing dotnet_home on helix to see if that works first

Comment (HaoK) [reply eng/helix/content/runtests.cmd:20]: actually I'll just set it to the DOTNET_ROOT since that might be less impactful

Comment (HaoK) [reply eng/helix/content/runtests.cmd:19]: We don't actually set DOTNET_HOME on the non windows versions either...(runtests.sh)

File: eng/helix/helix.proj
Comment (MattGal) [reply eng/helix/helix.proj:28]: @HaoK  yes, not until it is merged and flows out.

Comment (HaoK) [reply eng/helix/helix.proj:28]: Yup, that's what I thought, so I'll keep the temporary changes here for now since that lets us prep the other changes, and I'll back them out and rebase once the arcade changes are in

File: eng/helix/content/runtests.cmd
Comment (HaoK) [reply eng/helix/content/runtests.cmd:17]: @jkotalik @Tratcher progress, looks like the tests are passing now, but it does seem like there's a sporadic hang (with a dump attached) not sure if this is new or expected from time to time?

File: eng/helix/helix.proj
Comment (dougbu) [reply eng/helix/helix.proj:33]: Now that @alexperovich got dotnet/arcade#5748 in, it won't be much longer before we can get that fix. Importantly his design changed slightly and you'll need the `<DotNetCliPackageType>sdk</DotNetCliPackageType>` line. Will still be able to remove the `<Import>` line and `$(DotNetCliVersion)` setting.

File: eng/helix/content/runtests.cmd
Comment (dougbu) [inline eng/helix/content/runtests.cmd:15]: This comment isn't relevant anymore; no more `shift`

Comment (HaoK) [reply eng/helix/content/runtests.cmd:15]: I left it for posterity since when we add another argument, we will need to shift to get more still

**Comment by HaoK** [comment]
Status of this PR is waiting for the arcade changes to propogate, and then rebase to see if things go green (NetCoreSDK isn't set right now so runs are all broken)

**Comment by JunTaoLuo** [comment]
What arcade changes is this waiting on? Is this blocked on https://github.com/dotnet/aspnetcore/pull/23978?

**Comment by dougbu** [comment]
> Is this blocked on #23978?

Yes

**Comment by JunTaoLuo** [comment]
FYI this should be unblocked.

**Comment by HaoK** [comment]
Nice, with the new arcade changes, looks like the built in helix sdk support works for us now, marking as ready for review

File: eng/helix/content/runtests.cmd
Comment (dougbu) [inline eng/helix/content/runtests.cmd:18]: In our previous conversations about this change and about incorrectly changing the payload folders, I got the impression we were getting out from under the payload folders with this PR. Won't we hit leftover files still❔

I'm hoping this is fine and that I misunderstood you and @MattGal 😈

**Review by dougbu — APPROVED** [review]
Change looks fine. Just not positive whether it improves one area.

File: eng/helix/content/runtests.cmd
Comment (HaoK) [reply eng/helix/content/runtests.cmd:18]: If we want to install the runtime to a different directory we can still do that in a different PR, that's orthogonal to using the sdk support

File: eng/targets/Helix.targets
Comment (MattGal) [inline eng/targets/Helix.targets:120]: Suggestion:  Typically this would be expressed like:
`Condition=" '$(IsWindowsHelixQueue)' == 'true' ">` or `Condition=" '$(IsWindowsHelixQueue)' != 'true' ">`

**Review by MattGal — APPROVED** [review]

File: eng/targets/Helix.targets
Comment (dougbu) [reply eng/targets/Helix.targets:120]: @MattGal that's true when the property defaults to empty. But, when a property is always set to `true` or `false`, using it directly as a `bool` is easier to read and 💯% supported in `msbuild` and `dotnet msbuild`.

Comment (dougbu) [reply eng/targets/Helix.targets:120]: https://github.com/dotnet/aspnetcore/blob/324a90089f1eab71b5b8bae08e7f8cd6adaba1e9/eng/targets/Helix.props#L17-L18

**Comment by Pilchie** [comment]
:eyes:

**Pull request merged**
